py/bitbox02/bitbox02/bitbox02/secp256k1.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 Shift Cryptosecurity AG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""secp256k1 util functions"""
+import hashlib
+
+import ecdsa
+
+
+class ECDSANonceException(Exception):
+    pass
+
+
+def tagged_sha256(tag: bytes, msg: bytes) -> bytes:
+    tag_hash = hashlib.sha256(tag).digest()
+    return hashlib.sha256(tag_hash + tag_hash + msg).digest()
+
+
+def antiklepto_host_commit(host_nonce: bytes) -> bytes:
+    return tagged_sha256(b"s2c/ecdsa/data", host_nonce)
+
+
+def antiklepto_verify(host_nonce: bytes, signer_commitment: bytes, signature: bytes) -> None:
+    """
+    Verifies that hostNonce was used to tweak the nonce during signature
+    generation according to k' = k + H(signerCommitment, hostNonce) by checking that
+    k'*G = signerCommitment + H(signerCommitment, hostNonce)*G.
+    Throws ECDSANonceException if the verification fails.
+    """
+    assert len(host_nonce) == 32
+    assert len(signer_commitment) == 33, "expected uncompressed pubkey"
+    assert len(signature) == 64
+    signer_commitment_pubkey = ecdsa.VerifyingKey.from_string(
+        signer_commitment, ecdsa.curves.SECP256k1
+    )
+    # Compute R = R1 + H(R1, host_nonce)*G. R1 is the client nonce commitment, which we get
+    # uncompressed, but need to hash using the compressed serialization.
```

Comment:
`which we get uncompressed` is not accurate and from an earlier version. it is in the compressed form from the start.

Reply:
:pray: